va/va.go

```diff
@@ -374,7 +384,14 @@ func (va VAImpl) validateTLSALPN01(task *vaTask) *core.ValidationRecord {
 	leafCert := certs[0]
 
 	// Verify SNI - certificate returned must be issued only for the domain we are verifying.
-	if len(leafCert.DNSNames) != 1 || !strings.EqualFold(leafCert.DNSNames[0], task.Identifier) {
+	namematch := false
```

Comment:
nit: I usually prefer to have the implicit default instead of an explicit initialization for `bool` vars that will be flipped in later stages of execution.
```suggestion
	var namematch bool
```